lib/tree.rb

```diff
@@ -1,42 +1,74 @@
 class NoApplesError < StandardError; end
 
 class AppleTree
-  attr_#fill_in :height, :age, :apples, :alive
+  attr_accessor :height, :age, :apples, :alive
 
   def initialize
+    @height = 0
+    @age = 0
+    @apples = []
+    @alive = true
   end
 
   def age!
+    if alive
+      @age += 1
+      @height += 1 unless height >= 20
+      add_apples
+    end
+
+    @alive = false if age >= 100
   end
 
   def add_apples
+    apples_grown = age * 3
+
+    apples_grown.times do
+      @apples << new_apple
+    end
+  end
+
+  def new_apple
+    random_diameter = 2 + rand(4)
+    Apple.new('red', random_diameter)
   end
 
   def any_apples?
+    apples.any?
   end
 
   def pick_an_apple!
-    raise NoApplesError, "This tree has no apples" unless self.any_apples?
+    raise NoApplesError, 'This tree has no apples' unless any_apples?
+    @apples.pop
   end
 
   def dead?
+    alive == false
+  end
+end
+
+class Tree < AppleTree
+  def initialize
+    super
```

Comment:
Only define methods in the subclass that behave differently than the superclass.